yaml/secret.go

```diff
@@ -24,6 +24,15 @@ type (
 		Key    string `yaml:"key,omitempty"`
 		Engine string `yaml:"engine,omitempty"`
 		Type   string `yaml:"type,omitempty"`
+		Origin Origin `yaml:"origin,omitempty"`
```

Comment:
Could we add the `json` field tags here like we have for some other pipeline structs?